### PR TITLE
Fix hang in HVT_HYPERCALL_POLL on Linux

### LINKED ISSUE #460: solo5_yield()/HVT_HYPERCALL_POLL hangs on Linux if deadline has passed
While pairing with @Reperator yesterday, we discovered some inconvenience in respect to yield. On Linux with hvt, a tiny sleep on the MirageOS side never returns. This behaviour is different on FreeBSD. EDIT: The behaviour on spt aligns with the expected one (the timerfd uses `SYS_TFD_TIMER_ASTIME`).

An example unikernel (copied hello, adapted the sleep to 0 nanoseconds)

config.ml:
```OCaml
open Mirage
let main = foreign "Unikernel.Hello" (time @-> job)
let () = register "hello" [main $ default_time]
```

unikernel.ml:
```OCaml
open Lwt.Infix
module Hello (Time : Mirage_time.S) = struct
  let start _time =
    let rec loop = function
      | 0 -> Lwt.return_unit
      | n ->
        Logs.info (fun f -> f "hello");
        Time.sleep_ns 0L >>= fun () ->
        loop (n-1)
    in
    loop 4
end
```

expected behaviour (observed on FreeBSD with hvt):
```bash
            |      ___|
  __|  _ \  |  _ \ __ \
\__ \ (   | | (   |  ) |
____/\___/ _|\___/____/
Solo5: Bindings version v0.6.4
Solo5: Memory map: 512 MB addressable:
Solo5:   reserved @ (0x0 - 0xfffff)
Solo5:       text @ (0x100000 - 0x1d7fff)
Solo5:     rodata @ (0x1d8000 - 0x203fff)
Solo5:       data @ (0x204000 - 0x2a2fff)
Solo5:       heap >= 0x2a3000 < stack < 0x20000000
2020-05-12 09:06:39 -00:00: INF [application] hello
2020-05-12 09:06:39 -00:00: INF [application] hello
2020-05-12 09:06:39 -00:00: INF [application] hello
2020-05-12 09:06:39 -00:00: INF [application] hello
Solo5: solo5_exit(0) called
```

observed behaviour on Linux:
```bash
            |      ___|
  __|  _ \  |  _ \ __ \
\__ \ (   | | (   |  ) |
____/\___/ _|\___/____/
Solo5: Bindings version v0.6.4
Solo5: Memory map: 512 MB addressable:
Solo5:   reserved @ (0x0 - 0xfffff)
Solo5:       text @ (0x100000 - 0x1cffff)
Solo5:     rodata @ (0x1d0000 - 0x201fff)
Solo5:       data @ (0x202000 - 0x296fff)
Solo5:       heap >= 0x297000 < stack < 0x20000000
2020-05-12 09:08:55 -00:00: INF [application] hello
..and now it waits (indefinitely? - at least more than 10 seconds)..
```

### PR DESCRIPTION
**Pull request by mato**
Fixes #460, plus improvements to test_time.

### CODE REVIEW

**Comment by mato** [comment]
**15-basic-x86_64-OpenBSD67** is actually passing, looks like I need to downgrade node.js to 10.x there to stop surf-build throwing a fit every so often.

File: tenders/hvt/hvt_core.c [hunk 1]
```diff
@@ -224,6 +224,7 @@ static void hypercall_poll(struct hvt *hvt, hvt_gpa_t gpa)
             .tv_nsec = t->timeout_nsecs % 1000000000ULL
         }
     };
+    it.it_value.tv_nsec |= 1;
```
Comment (hannesm) [inline tenders/hvt/hvt_core.c:227]: I recommend a comment for this line, it is non-obvious why it is here (and not in the other cases) - until you read up on `timerfd_settime`.

Comment (mato) [reply tenders/hvt/hvt_core.c:232]: It's in the commit history, but OK, done.

**Comment by mato** [comment]
Ok, I think this looks good, I'll quit poking at `test_time` for now and do (more of) that at some later time.

**Pull request merged**